entry_types/scrolled/package/spec/contentElements/dataWrapperChart/useIframeHeight-spec.js

```diff
@@ -0,0 +1,19 @@
+import {useIframeHeight} from '../../../src/contentElements/dataWrapperChart/useIframeHeight'
+
+import {renderHook} from '@testing-library/react-hooks';
+import {fakeParentWindow, tick} from 'support';
+import {act} from '@testing-library/react';
+
+describe('useIframeHeight', () => {
```

Comment:
Consider adding a spec that posts a message with the expected format, but leads to a width that is different from 400px, using `postMessage`. I see two advantages:

- spec tests more of the hook's functionality than it currently does
- spec contains an example of how we'd expect message payload to look like, which makes it easier for us to adapt in case Datawrapper's payload changes

Reply:
I changed the functionality of the hook to accommodate the tests. Now we can check that if the value changes then the listener is called with the new value and changes the state.

Reply:
I wouldn't change the hook for that. The previous interface was leaner and better. Consider spying on `window.document`'s `removeEventListener` in the spec and reverting the changes to the hook

Reply:
Now, the spec that posted the `({'datawrapper-height': {id : 350}}` message seems to have been reverted as well. That spec was fine. Could you re-add that?

Reply:
Very nice now! Also thanks to @awaismbajwa for taking a look!